benchmarks/diffusion_2D/diffusion_2D.hpp

```diff
@@ -29,6 +29,8 @@
 
 #include "mpi.h"
 
+#include <sundials/sundials_core.hpp>
```

Comment:
It is best practice to include the header files that you use names from in the file you use them rather than relying on transitive includes. With that in mind, either the individual header files inside of sundials_core.h/sundials_core.hpp should be included as needed or sundials_core.h/sundials_core.hpp should be included. The point of creating sundials_core.h/sundials_core.hpp was to simplify things for the user while still supporting the "include what you use" principle in spirit via an idiom (aggregating header files) that is common to many C/C++ libraries.